mesonbuild/utils/universal.py

```diff
@@ -756,40 +757,54 @@ def windows_detect_native_arch() -> str:
             raise EnvironmentException('Unable to detect native OS architecture')
     return arch
 
-def detect_vcs(source_dir: T.Union[str, Path]) -> T.Optional[T.Dict[str, str]]:
+@dataclasses.dataclass
+class VcsData:
+    name: str
+    cmd: str
+    repo_dir: str
+    get_rev: 'T.List[str]'
```

Comment:
There's no need to quote these, `__future__.annotations` takes care of that for us.